bitcoin/psbt.c

```diff
@@ -401,40 +386,47 @@ void psbt_elements_normalize_fees(struct wally_psbt *psbt)
 			psbt_rm_output(psbt, i--);
 			continue;
 		}
-		if (!amount_asset_is_main(&asset))
+		if (!amount_asset_is_main(&output_amount))
 			continue;
 
-		if (!amount_sat_sub(&total_in, total_in,
-				    amount_asset_to_sat(&asset)))
+		if (!amount_sat_sub(&total_fee, total_fee,
+				    amount_asset_to_sat(&output_amount)))
 			return;
 	}
 
-	if (amount_sat_eq(total_in, AMOUNT_SAT(0)))
+	if (amount_sat_eq(total_fee, AMOUNT_SAT(0)))
 		return;
 
 	/* We need to add a fee output */
 	if (fee_output_idx == psbt->num_outputs) {
-		psbt_append_output(psbt, NULL, total_in);
+		psbt_append_output(psbt, NULL, total_fee);
 	} else {
-		u64 sats = total_in.satoshis; /* Raw: wally API */
-		struct wally_tx_output *out = &psbt->tx->outputs[fee_output_idx];
-		if (wally_tx_confidential_value_from_satoshi(
-			sats, out->value, out->value_len) != WALLY_OK)
+		u64 sats = total_fee.satoshis; /* Raw: wally API */
+		struct wally_psbt_output *out = &psbt->outputs[fee_output_idx];
+		if (wally_psbt_output_set_amount(out, sats) != WALLY_OK) {
 			return;
```

Comment:
I realize this is just the previous code, but this seems awkward.  Returning here seems like a fail?  How can this fail?  Can we just assert?